internal/server/events_test.go

```
@@ -286,10 +277,129 @@ func TestSSENonBlockingSend(t *testing.T) {
+
+func TestDirChangedEndpointEmitsEvent(t *testing.T) {
+	dir := t.TempDir()
+	os.MkdirAll(filepath.Join(dir, "a"), 0o755)
+
+	hub := NewEventHub(dir, nil, nil)
+	hub.Start()
+	defer hub.Stop()
+
+	srv := &Server{root: dir, events: hub}
+
+	req := httptest.NewRequest("GET", "/events", nil)
```

golangci-lint (2.11.4)

[error] net/http/httptest.NewRequest must not be called. use net/http/httptest.NewRequestWithContext

(noctx)